opendevin/runtime/plugins/jupyter/execute_server

```diff
@@ -71,6 +71,8 @@ class JupyterKernel:
         self.tools_to_run = [
             # TODO: You can add code for your pre-defined tools here
         ]
+        if os.path.exists('/opendevin/plugins/agent_skills/agentskills.py'):
```

Comment:
This line confuses me a bit. Why do we need this?

Reply:
There might be cases where the "jupyter" plugin is enabled, but NOT `agentskills` so we need to add this check.  Eventually I'm actually hoping to combine `agentskills` and `jupyter` into ONE unified `plugin` (so we don't need this check).

Reply:
Oh I see. Shouldn't we throw an exception or something? If we don't have the toolkit installed but the LLM attempts to use it, then we are screwed...?

Reply:
I guess there could potentially be a case where the LLM will need to use `jupyter` without `agentskill` (though i guess it is probably pretty unlikely 

-- But I'm thinking of merging the entire `jupyter` thing into the `agentskill` library too. Maybe we can just do this, so that we don't need to worry about this anymore. Maybe in a follow-up PR once this is merged to make things cleaner?

Reply:
Maybe I'm misunderstanding, feel free to correct me but TBH I can see an argument for keeping it this way, including in the kernel reload... We could add more things/plugins to the Jupyter kernel, and they don't all have to be mandatory? Having a template how to do it is appreciated, too ❤️ 

Lately I've been playing with adding opendevin dependencies in Jupyter environment, and FWIW if the LLM attempts something and it doesn't work, usually it doesn't seem a problem. Until it decides to kill its own Jupyter, all good! 😅